types/mapbox__mapbox-gl-draw/index.d.ts

```diff
@@ -159,51 +159,65 @@ declare namespace MapboxDraw {
     }
 
     interface DrawCustomMode<CustomModeState = any, CustomModeOptions = any> {
-        onSetup?(this: DrawCustomModeThis, options: CustomModeOptions): CustomModeState;
+        onSetup?(this: DrawCustomModeThis & this, options: CustomModeOptions): CustomModeState;
```

Comment:
Hm, this is a bit of an awkward pattern, but I can see why you would need it.

Reply:
Yep, it is awkward pattern, but it's really needed